cmd/syncthing/gui.go

```diff
@@ -400,24 +432,50 @@ func (s *apiService) String() string {
 }
 
 func (s *apiService) VerifyConfiguration(from, to config.Configuration) error {
-	_, err := net.ResolveTCPAddr("tcp", to.GUI.Address())
-	return err
+	for _, guiCfg := range to.GUIs() {
+		if _, err := net.ResolveTCPAddr("tcp", guiCfg.Address); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func (s *apiService) CommitConfiguration(from, to config.Configuration) bool {
-	// No action required when this changes, so mask the fact that it changed at all.
-	from.GUI.Debugging = to.GUI.Debugging
+	configChanged := false
+	for _, toGuiCfg := range to.GUIs() {
+		fromGuiCfg := from.GUIByAddress(toGuiCfg.Address)
+		if fromGuiCfg == nil {
+			// A new GUI configuration has been added.
+			configChanged = true
+			break
+		}
 
-	if to.GUI == from.GUI {
-		return true
-	}
+		// No action required when this changes, so mask the fact that it changed at all.
+		fromGuiCfg.Debugging = toGuiCfg.Debugging
+
+		if toGuiCfg == *fromGuiCfg {
+			continue
+		}
+
+		if toGuiCfg.Theme != fromGuiCfg.Theme {
+			s.staticsMap[toGuiCfg.Address].setTheme(toGuiCfg.Theme)
+		}
 
-	if to.GUI.Theme != from.GUI.Theme {
-		s.statics.setTheme(to.GUI.Theme)
+		configChanged = true
 	}
+	for _, fromGuiCfg := range from.GUIs() {
+		toGuiCfg := to.GUIByAddress(fromGuiCfg.Address)
+		if toGuiCfg == nil {
+			// A GUI configuration has been removed. GUI restart is needed.
+			return false
```

Comment:
Wouldn't `configChanged=true` be enough? I don't think a full restart of Syncthing is needed, rerunning `Serve` is enough, isn't it?

Reply:
Yeah, service restart would rewind all UIs

Reply:
Fine